packages/builder/plugin-image-compress/src/index.ts

```diff
@@ -1,19 +1,46 @@
 import type { BuilderPlugin } from '@modern-js/builder-webpack-provider/types';
+import assert from 'assert';
 import { ModernJsImageMinimizerPlugin } from './minimizer';
 import { withDefaultOptions } from './shared/utils';
 import { Codecs, Options } from './types';
 
 export type PluginImageCompressOptions = Options[];
 export const DEFAULT_OPTIONS: Codecs[] = ['jpeg', 'png', 'ico'];
 
+export interface IPluginImageCompress {
+  (...options: Options[]): BuilderPlugin;
+  (options: Options[]): BuilderPlugin;
+}
+
+const castOptions = (args: (Options | Options[])[]): Options[] => {
+  const head = args[0];
+  // expect [['png', { use: 'jpeg' }]]
+  if (Array.isArray(head)) {
+    return head;
+  }
+  // expect ['png', { use: 'jpeg' }]
+  const ret: Options[] = [];
+  for (const arg of args) {
+    assert(!Array.isArray(arg));
```

Comment:
why do we need to assert the arg

Reply:
Just for typing, not necessary. TypeScript can't distinguish between the two subtypes of the union type.